### PR TITLE
Properly declare missing properties on classes

### PR DESCRIPTION
**Pull request by renatonascalves**
Trac ticket: https://buddypress.trac.wordpress.org/ticket/7018

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

File: src/bp-groups/classes/class-bp-groups-group-members-template.php [hunk 1]
```diff
@@ -64,6 +64,21 @@ class BP_Groups_Group_Members_Template {
...
+	 *
+	 * @since 1.0.0
+	 * @var string
+	 */
+	public $pag_arg;
+
+	/**
+	 * The total number of members.
+	 *
+	 * @var int
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-group-members-template.php:78]: `int` => `integer` ?

File: src/bp-groups/classes/class-bp-groups-invite-template.php [hunk 2]
```diff
@@ -40,6 +40,14 @@ class BP_Groups_Invite_Template {
...
+	 * List of invites found and their respective data.
+	 *
+	 * @since 1.1.0
+	 * @var array
+	 */
+	public $invite_data = array();
+
 	/**
 	 * @since 1.1.0
 	 * @var bool
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-invite-template.php:53]: `bool` => `boolean` ?

File: src/bp-groups/classes/class-bp-groups-invite-template.php [hunk 3]
```diff
@@ -64,6 +72,14 @@ class BP_Groups_Invite_Template {
...
+	 * URL argument used for the pagination param.
+	 *
+	 * @since 1.1.0
+	 * @var string
+	 */
+	public $pag_arg;
+
 	/**
 	 * @since 1.1.0
 	 * @var int
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-invite-template.php:85]: `int` => `integer` ?

File: src/bp-groups/classes/class-bp-groups-membership-requests-template.php [hunk 4]
```diff
@@ -64,6 +64,14 @@ class BP_Groups_Membership_Requests_Template {
...
+	 * URL argument used for the pagination param.
+	 *
+	 * @since 1.0.0
+	 * @var string
+	 */
+	public $pag_arg;
+
 	/**
 	 * @since 1.0.0
 	 * @var int
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-membership-requests-template.php:77]: `int` => `integer` ?

File: src/bp-groups/classes/class-bp-groups-template.php [hunk 5]
```diff
@@ -21,96 +21,104 @@ class BP_Groups_Template {
 	/**
 	 * The loop iterator.
 	 *
-	 * @var int
 	 * @since 1.2.0
+	 * @var int
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-template.php:25]: `int` => `integer` ?

File: src/bp-groups/classes/class-bp-groups-template.php [hunk 6]
```diff
@@ -21,96 +21,104 @@ class BP_Groups_Template {
...
 	 */
 	public $current_group = -1;
 
 	/**
 	 * The number of groups returned by the paged query.
 	 *
-	 * @var int
 	 * @since 1.2.0
+	 * @var int
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-template.php:33]: `int` => `integer` ?

File: src/bp-groups/classes/class-bp-groups-template.php [hunk 7]
```diff
@@ -21,96 +21,104 @@ class BP_Groups_Template {
...
+	 * @var object
 	 */
 	public $group;
 
 	/**
 	 * A flag for whether the loop is currently being iterated.
 	 *
-	 * @var bool
 	 * @since 1.2.0
+	 * @var bool
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-template.php:57]: `bool` => `boolean` ?

File: src/bp-groups/classes/class-bp-groups-template.php [hunk 8]
```diff
@@ -21,96 +21,104 @@ class BP_Groups_Template {
...
+	 * @var string
 	 */
 	public $pag_links;
 
 	/**
 	 * The total number of groups matching the query parameters.
 	 *
-	 * @var int
 	 * @since 1.2.0
+	 * @var int
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-template.php:97]: `int` => `integer` ?

File: src/bp-groups/classes/class-bp-groups-template.php [hunk 9]
```diff
@@ -21,96 +21,104 @@ class BP_Groups_Template {
...
 	 */
 	public $total_group_count;
 
 	/**
 	 * Whether the template loop is for a single group page.
 	 *
-	 * @var bool
 	 * @since 1.2.0
+	 * @var bool
```
Comment (imath) [inline src/bp-groups/classes/class-bp-groups-template.php:105]: `bool` => `boolean` ?

File: src/bp-messages/classes/class-bp-messages-box-template.php [hunk 10]
```diff
@@ -36,6 +36,13 @@ class BP_Messages_Box_Template {
 	 */
 	public $total_thread_count = 0;
 
+	/**
+	 * Total number of threads available to iterate on.
+	 *
+	 * @var int
```
Comment (imath) [inline src/bp-messages/classes/class-bp-messages-box-template.php:42]: `int` => `integer` ?

File: src/bp-notifications/classes/class-bp-notifications-template.php [hunk 11]
```diff
@@ -75,6 +75,22 @@ class BP_Notifications_Template {
 	 */
 	public $user_id;
 
+	/**
+	 * The status of the notification.
+	 *
+	 * @since 1.9.0
+	 * @var bool
```
Comment (imath) [inline src/bp-notifications/classes/class-bp-notifications-template.php:82]: `bool` => `boolean` ?

File: src/bp-notifications/classes/class-bp-notifications-template.php [hunk 12]
```diff
@@ -75,6 +75,22 @@ class BP_Notifications_Template {
...
+	 */
+	public $is_new;
+
+	/**
+	 * The total notification count.
+	 *
+	 * @since 1.9.0
+	 * @var int
```
Comment (imath) [inline src/bp-notifications/classes/class-bp-notifications-template.php:90]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 13]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
 	 * Field ID.
 	 *
 	 * @since 1.0.0
-	 * @var int ID of field.
+	 * @var int
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:24]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 14]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
 	 */
 	public $id;
 
 	/**
 	 * Field group ID.
 	 *
 	 * @since 1.0.0
-	 * @var int Field group ID for field.
+	 * @var int
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:32]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 15]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
 	 */
 	public $group_id;
 
 	/**
 	 * Field parent ID.
 	 *
 	 * @since 1.0.0
-	 * @var int Parent ID of field.
+	 * @var int
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:40]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 16]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
+	 * @var string
 	 */
 	public $description;
 
 	/**
 	 * Required field?
 	 *
 	 * @since 1.0.0
-	 * @var bool Is field required to be filled out?
+	 * @var bool
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:72]: `bool` => `boolean` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 17]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
 	 */
 	public $is_required;
 
 	/**
 	 * Deletable field?
 	 *
 	 * @since 1.0.0
-	 * @var int Can field be deleted?
+	 * @var int
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:80]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 18]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
 	 */
 	public $can_delete = '1';
 
 	/**
 	 * Field position.
 	 *
 	 * @since 1.0.0
-	 * @var int Field position.
+	 * @var int
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:88]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 19]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
 	 */
 	public $field_order;
 
 	/**
 	 * Option order.
 	 *
 	 * @since 1.0.0
-	 * @var int Option order.
+	 * @var int
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:96]: `int` => `integer` ?

File: src/bp-xprofile/classes/class-bp-xprofile-field.php [hunk 20]
```diff
@@ -21,104 +21,112 @@ class BP_XProfile_Field {
...
+	 * @var string
 	 */
 	public $order_by;
 
 	/**
 	 * Is this the default option?
 	 *
 	 * @since 1.0.0
-	 * @var bool Is this the default option for this field?
+	 * @var bool
```
Comment (imath) [inline src/bp-xprofile/classes/class-bp-xprofile-field.php:112]: `bool` => `boolean` ?

**Review by imath — COMMENTED** [review]
Looks good I just have questions about whether we should use bool/int or boolean/integer everywhere ? Thanks for your work 💪

**Comment by renatonascalves** [comment]
@imath Initially, I was updating the types (i.e., `bool` => `boolean`) per case/situation, but I'll introduce a future pull request with a change in the project. In this way, we account for everything.

**Pull request closed without merging**